### PR TITLE
[Improvement] Enhance alignment of metadata fields

### PR DESCRIPTION
**Pull request by s7monk**
### Purpose

The vo subsection of the metadata is currently incomplete. For example, DataFileVO lacks many fields. This PR completes the fields of VO.

### CODE REVIEW

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotVO.java [hunk 1]
```diff
@@ -18,72 +18,47 @@
...
+
+    private String changelogManifestList;
+
+    private Long totalRecordCount;
+
+    private Long deltaRecordCount;
+
+    private Long changelogRecordCount;
+
+    private Integer addedFileCount;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotVO.java:59]:
```suggestion
    private Long addedFileCount;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotVO.java [hunk 2]
```diff
@@ -18,72 +18,47 @@
...
+
+    private Integer deletedFileCount;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotVO.java:61]:
```suggestion
    private Long deletedFileCount;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 3]
```diff
@@ -108,11 +108,25 @@ public List<SnapshotVO> getSnapshot(MetadataDTO dto) {
...
+                                        .commitIdentifier(internalRow.getLong(3))
+                                        .commitKind(internalRow.getString(4).toString())
+                                        .commitTime(
                                                 internalRow.getTimestamp(5, 3).toLocalDateTime())
+                                        .baseManifestList(internalRow.getString(6).toString())
+                                        .deltaManifestList(internalRow.getString(7).toString())
+                                        .changelogManifestList(
+                                                internalRow.getString(8) != null
+                                                        ? internalRow.getString(8).toString()
+                                                        : "")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:123]: Why not null?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 4]
```diff
@@ -157,11 +173,32 @@ public List<DataFileVO> getDataFile(MetadataDTO dto) {
...
+                                        .filePath(internalRow.getString(2).toString())
+                                        .fileFormat(internalRow.getString(3).toString())
+                                        .schemaId(internalRow.getLong(4))
+                                        .level(internalRow.getInt(5))
+                                        .recordCount(internalRow.getLong(6))
+                                        .fileSizeInBytes(internalRow.getLong(7))
+                                        .minKey(
+                                                internalRow.getString(8) != null
+                                                        ? internalRow.getString(8).toString()
+                                                        : "")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:189]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 5]
```diff
@@ -157,11 +173,32 @@ public List<DataFileVO> getDataFile(MetadataDTO dto) {
...
+                                        .maxKey(
+                                                internalRow.getString(9) != null
+                                                        ? internalRow.getString(9).toString()
+                                                        : "")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:193]: Ditto.

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@s7monk, thanks for contribution. Please add the test case for this pull request.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java (hunk 3 above)
Comment (s7monk) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:123]: During the test, it was found that internalRow.getString(8) may return null, and if no judgment is made, a null pointer will be reported.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java (hunk 4 above)
Comment (s7monk) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:189]: During the test, it was found that internalRow.getString(8) may return null, and if no judgment is made, a null pointer will be reported.

**Comment by s7monk** [comment]
> @s7monk, thanks for contribution. Please add the test case for this pull request.

I just took a look. The current test only tests whether the interface can be requested, but does not test specific fields, so adding fields here does not require adding a test.

**Comment by s7monk** [comment]
@SteNicholas PTAL.

**Comment by s7monk** [comment]
> > @s7monk, thanks for contribution. Please add the test case for this pull request.
> 
> I just took a look. The current test only tests whether the interface can be requested, but does not test specific fields, so adding fields here does not require adding a test.

I have verified whether the logic is correct through another PR (https://github.com/apache/incubator-paimon-webui/pull/150), After merging, I will add a test to this PR.

**Pull request closed without merging**